### ISSUE TITLE
Token Manager "removed " event do not emit when tokens stored in cookies are removed from another application

### ISSUE THREAD

**Issue by boubou158** [okta/okta-auth-js#1391]
### Describe the bug?

Hi,

I am opening this post following the solution used for https://github.com/okta/okta-auth-js/issues/1387.

Since ID token/ Access token for application A are stored in the cookies, and getting deleted when a logout from application B occurs, I would expect the tokenManager on application A to trigger a "removed" event. But it does not seem to be the case.

What do you suggest for this? Do you think adding in the code of application A, a timer and update tokenManager state by calling tokenManager.clear() if tokens are not here anymore would do the trick?

```
interval(10000).subscribe(async () => {
      const tokens = await this.oktaAuth.tokenManager.getTokens();
      if (!Object.keys(tokens).length) {
        this.oktaAuth.tokenManager.remove('idToken');
        this.oktaAuth.tokenManager.remove('accessToken');
      }
    });
```

Thanks in advance :)

### What is expected to happen?

NA

### What is the actual behavior?

NA

### Reproduction Steps?

NA

### SDK Versions

NA

### Execution Environment

NA

### Additional Information?

NA

**Comment by denysoblohin-okta** [comment]
What version of `okta-auth-js` do you use?

Tokens synchronisation across tabs should be [enabled by default](https://github.com/okta/okta-auth-js#services) with [syncStorage](https://github.com/okta/okta-auth-js#syncstorage-1).
However it works only within one application (client id) by design, so it will not help with your issue. But you can use similar approach.

You can subscribe to [`removed`](https://github.com/okta/okta-auth-js#tokenmanageronevent-callback-context) event for `TokenManager` instance in app1 and post custom message to another tab running app2.
We use [broadcast-channel](https://www.npmjs.com/package/broadcast-channel) for cross-tab communication in our SDK, see https://github.com/okta/okta-auth-js/blob/master/lib/services/SyncStorageService.ts#L107
We generate [channel name](https://github.com/okta/okta-auth-js/blob/4fe01df8b53435fe6935e54fbaa625242ea8a922/lib/core/ServiceManager/browser.ts#L66) using client id, but you should use some more general name to enable cross-app communication.

**Comment by boubou158** [comment]
Hi @denysoblohin-okta ,

In our case, cookies okta-token-storage_idToken and okta-token-storage_accessToken are cleared from a backend call which does a 
Set-Cookie: okta-token-storage_accessToken=deleted; Max-Age=0; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Domain=mydomain.com; Path=/; Secure; on both tokens to delete them from the browser.
It does not pass through the SDK for deletion hence no events "removed" are triggered. Maybe the SDK could check periodically every X seconds if the idTokens/ accessTokens disappeared and trigger the events if they got removed. (the solution provided in my initial message)

Best regards

**Comment by denysoblohin-okta** [comment]
You can use timer approach like in your example.

Or you can still use broadcast approach - broadcast `removed` custom message (eg. with broadcast-channel) after your backend request (which sets cookie) is completed, and listen to such event in other tabs.

**Comment by jaredperreault-okta** [comment]
tokens stored in cookies via `TokenManager` are designed to be managed purely by the `TokenManager`, which is why you're not getting any storage events. Is there a specific reason you're clearing cookies via `Set-Cookie` headers rather than using `TokenManager` methods?

**Comment by boubou158** [comment]
Hi @jaredperreault-okta ,

In fact, it would be easier to rely fully on the sdk on every applications. Unfortunately, we integrated our application with other applications which are not using the SDK and introduced a cookie based mecanism to trigger the SSO authentication. It added a lot of complexity but modifying the other applications to use the SDK was not an option.

It seems we reached a quiet stable solution at this step. Thanks again for all the help.

Best regards,
